setuptools/config/_validate_pyproject/formats.py

```diff
@@ -91,9 +91,9 @@ def pep508(value: str) -> bool:
         """
         try:
             _req.Requirement(value)
-            return True
         except _req.InvalidRequirement:
             return False
+        return True
```

Comment:
I actually prefer the form on the left, as it provides a more symmetric style. The form on the right entangles line 93 with 96 and 94 with 95. The control flow is irregular.

Even better, I've found the use of [`ExceptionTrap`](https://jaracocontext.readthedocs.io/en/latest/#jaraco.context.ExceptionTrap) to side-step these concerns and provide an even more elegant syntax:

```python
@ExceptionTrap(_req.InvalidRequirement).passes
def pep508(value: str): -> bool:
    _req.Requirement(value)
```

or even

```python
pep508 = ExceptionTrap(_req.InvalidRequirement).passes(_req.Requirement)
```

Or, if you're really feeling more like explicit is better than implicit:

```python
def pep508(value: str): -> bool:
    with ExceptionTrap(_req.InvalidRequirement) as ctx:
       _req.Requirement(value)
    return bool(ctx)
```

Reply:
I guess it's not about style, but applying `try:` only to the piece of code we expect to raise an exception, and not to other code that might raise totally unexpected exceptions. Of course, a mere `return` will never raise an exception in this specific case!

I prefer this, but I know some rules (not enforced here) complain about it.
```python
        try:
            _req.Requirement(value)
        except _req.InvalidRequirement:
            return False
        else:
            return True
```